src/pages/Analytics/Home/HomeAnalyticList.js

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect, useState } from 'react'
+import i18n from '@dhis2/d2-i18n'
+import PropTypes from '@dhis2/prop-types'
+import isEmpty from 'lodash/isEmpty'
+import isEqual from 'lodash/isEqual'
+import { HomeVisualizationTable } from '../../../components/analyticVisualization'
+import { AddNewSetting } from '../../../components/field'
+
+const HomeAnalyticList = ({
+    visualizations,
+    handleVisualizations,
+    disable,
+}) => {
+    const [rows, setRows] = useState()
+    const [initialRows, setInitialRows] = useState()
+
+    useEffect(() => {
+        if (visualizations) {
+            setRows(visualizations)
+            setInitialRows(visualizations)
+        }
+    }, [visualizations])
+
+    useEffect(() => {
+        if (rows && initialRows && !isEqual(rows, initialRows)) {
+            handleVisualizations(rows)
+        }
+    }, [rows])
```

Comment:
If `initialRows` and `visualizations` are always in sync, this should be simpler:

```suggestion
    useEffect(() => {
        if (visualizations) {
            setRows(visualizations)
        }
    }, [visualizations])

    useEffect(() => {
        if (rows && visualizations && !isEqual(rows, visualizations)) {
            handleVisualizations(rows)
        }
    }, [rows])
```